### ISSUE TITLE
java.lang.Thread.State: BLOCKED (on object monitor) at org.keycloak.adapters.rotat8.getPublicKey(JWKPublicKeyLocator.java:60)

### ISSUE THREAD

**Issue by samir-kamble** [keycloak/keycloak#22121]
### Before reporting an issue

- [x] I have searched existing issues
- [ ] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

authentication

### Describe the bug

Hello, 

We are using Keycloak v18 which is giving mentioned issue and hanging up JAVA services.

Thread is getting blocked by Keycloak function and hence JAVA services are getting hanged.

Please find complete jstack thread dump.

### Version

18

### Expected behavior

Thread must not block JAVA services.

### Actual behavior

Threads are blocking JAVA services and hanging them

### How to Reproduce?

NA

**Comment by rmartinc** [comment]
Hi @samir-kamble!

If you check that part of the code is a [synchronized section](https://github.com/keycloak/keycloak/blob/18.0.2/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/rotation/JWKPublicKeyLocator.java#L60-L70) which is isolated to load the JWKs from the URL configured. With the information you provide we cannot do much. I think that probably the thread is hanging inside the synch section trying to download the JWKs ([here](https://github.com/keycloak/keycloak/blob/18.0.2/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/rotation/JWKPublicKeyLocator.java#L63)). Check if the thread is hanging in the http exchange and if you can access that endpoint from the server machine.

Regards!

**Comment by mposolda** [comment]
@rmartinc Thanks for checking this.

I am adding to Backlog as adapters are in general deorecated and low priority and Keycloak team likely won't have time to look at this in the near future.

I would also doublecheck firewall settings. Sometimes I/O requests are stucked due the firewall (in this case firewall when adapter tries to download keys from the Keycloak server.)

**Comment by rmartinc** [comment]
I'm closing this as adapters are going to be removed in the next release and this is probably a network issue.
